src/main/java/net/glowstone/util/LibraryManager.java

```diff
@@ -133,9 +195,81 @@ String getLibrary() {
             return library + '-' + version + ".jar";
         }
 
-        boolean checksum(File file, String checksum) {
-            // TODO: actually check checksum
-            return file.exists();
+        /**
+         * Computes and validates the checksum of a file.
+         * <p>If the file does not exist, the checksum will be automatically invalidated.
+         * <p>If the reference checksum or the algorithm are empty or null,
+         * the checksum will be automatically validated.
+         *
+         * @param file the file.
+         * @param checksum the reference checksum to validate the file's digest with.
+         * @param algorithm the {@link HashAlgorithm} used to compute the file's digest.
+         * @return true if the checksum was validated, false otherwise.
+         */
+        boolean checksum(File file, String checksum, HashAlgorithm algorithm) {
+            checkNotNull(file);
+            if (!file.exists()) {
+                return false;
+            }
+            if (algorithm == null || checksum == null || checksum.isEmpty()) {
+                // assume everything is OK if no reference checksum is provided
+                return true;
+            }
+            // get the file digest
+            String digest;
+            try {
+                digest = Files.hash(file, algorithm.getFunction()).toString();
+            } catch (IOException ex) {
+                GlowServer.logger.log(Level.SEVERE, "Failed to compute digest for '" + file.getName() + "'", ex);
+                return false;
+            }
+            return digest.equals(checksum);
+        }
+    }
+
+    public enum HashAlgorithm {
+
+        SHA1(DEFAULT_HASH_FUNCTION, DEFAULT_HASH_FUNCTION_NAME),
```

Comment:
One comment here, I don't think DEFAULT_HASH_FUNCTION_NAME and ALTERNATIVE_HASH_FUNCTION_NAME should be static variables, since they're already properties of the enum. In fact, I would argue that ALTERNATIVE_HASH_FUNCTION should be gotten rid of as a static variable as well, since it'll make for some awkward code once a third hash is added.